src/structures/idt.rs

```diff
@@ -458,25 +461,21 @@ impl InterruptDescriptorTable {
         }
     }
 
-    /// Returns a normalized and ranged check slice range from a RangeBounds trait object
+    /// Returns a normalized and ranged check slice range from a RangeBounds trait object.
     ///
-    /// Panics if range is outside the range of user interrupts (i.e. greater than 255) or if the entry is an
-    /// exception
-    fn condition_slice_bounds(&self, bounds: impl RangeBounds<usize>) -> (usize, usize) {
+    /// Panics if the entry is an exception.
+    fn condition_slice_bounds(&self, bounds: impl RangeBounds<u8>) -> (usize, usize) {
         let lower_idx = match bounds.start_bound() {
-            Included(start) => *start,
-            Excluded(start) => *start + 1,
+            Included(start) => (*start as usize),
+            Excluded(start) => (*start as usize) + 1,
```

Comment:
Instead of `as` casts, I prefer `from()` for such lossless conversions to larger integer types:

```suggestion
            Included(start) => usize::from(*start),
            Excluded(start) => usize::from(*start) + 1,
```